app/controllers/incomeSources/add/IncomeSourceAddedController.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.add
+
+import auth.MtdItUser
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
+import config.featureswitch.FeatureSwitching
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.predicates.{AuthenticationPredicate, IncomeSourceDetailsPredicate, NavBarPredicate, NinoPredicate, SessionTimeoutPredicate}
+import enums.IncomeSourceJourney.{ForeignProperty, IncomeSourceType, SelfEmployment, UkProperty}
+import play.api.Logger
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{DateServiceInterface, IncomeSourceDetailsService, NextUpdatesService}
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import utils.IncomeSourcesUtils
+import views.html.incomeSources.add.IncomeSourceAddedObligations
+
+import java.time.LocalDate
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class IncomeSourceAddedController @Inject()(authenticate: AuthenticationPredicate,
```

Comment:
* you might need to mark this controller as 'Shared' as this is shared between few endpoints; thought personally I am against shared controllers and this seem to me as an 'anti-patterns', as we should be able to move shared logic into trait in the worst case scenarios and keep existing controller as they are;

Reply:
All the shared controllers are called 'IncomeSource"XX rather than Business/Foreign/UK, so I think this is clear already.

Reply:
this type of naming convention would be useful in the future, as I this would help us to identify such controller when we would need to de-compose these back into separate controllers ie.

Reply:
I guess this would be a separate ticket since it would affect many controllers other than this one. Also some controllers have been built as joint so never had separate controllers for different journeys (e.g. the manage journey)